codespeed/views.py

```diff
@@ -731,3 +739,239 @@ def add_json_results(request):
     logger.debug("add_json_results: completed")
 
     return HttpResponse("All result data saved successfully", status=202)
+
+
+def django_has_content_type():
+    return (django.VERSION[0] > 1 or
+            (django.VERSION[0] == 1 and django.VERSION[1] >= 6))
+
+
+def validate_results_request(data):
+    """
+    Validates that a result request dictionary has all needed parameters
+
+    It returns a tuple
+        "", False  when no errors where found
+        Error_message, True when there is an error
+    """
+    mandatory_data = [
+        'env',
+        'proj',
+        'branch',
+        'exe',
+        'ben',
+    ]
+
+    for key in mandatory_data:
+        if key not in data:
+            return 'Key "' + key + '" missing from GET request', True
+        elif data[key] == '':
+            return 'Value for key "' + key + '" empty in GET request', True
+
+    # Check that 'revs' is the correct format (if it exists)
+    if 'revs' in data:
+        try:
+            rev_value = int(data['revs'])
+        except:
+            return 'Value for key "revs" is not an integer', True
+        if rev_value <= 0:
+            return 'Value for key "revs" should be a strictly positive '\
+                   'integer', True
+
+    return '', False
+
+
+def get_benchmark_results(data):
```

Comment:
This could could go in `views_data.py`, where the other functions to fetch data for the views are.